src/js/services/integrations-service.js

```diff
@@ -63,10 +63,17 @@ export function linkTwilioChannel(userId, data) {
                 linkState: 'pending'
             });
         })
-        .catch(() => {
+        .catch((e) => {
+            const {response: {status}} = e;
+            let errorMessage;
+            if (status === 429) {
+                errorMessage = 'A link was attempted recently for this phone number. Please wait before trying again.';
+            } else {
+                errorMessage = 'We were unable to communicate with this number. Please enter a different one.';
```

Comment:
If a `400` is returned this is the appropriate error, but if it's `500` maybe a try again might be good